### ISSUE TITLE
Only 1 deformable convolution

### ISSUE THREAD

**Issue by bkvie** [ChunhuanLin/deform_conv_pytorch#4]
```
 self.conv2 = nn.Conv2d(32, 64, kernel_size=3, padding=1)
    self.bn2 = nn.BatchNorm2d(64)

    self.conv3 = nn.Conv2d(64, 128, kernel_size=3, padding=1)
    self.bn3 = nn.BatchNorm2d(128)

    self.offsets = nn.Conv2d(128, 18, kernel_size=3, padding=1)
    self.conv4 = **DeformConv2D**(128, 128, kernel_size=3, padding=1)
    self.bn4 = nn.BatchNorm2d(128)

    self.classifier = nn.Linear(128, 10)

```

Why are you only using 1 DefConv layer? Why not make all of them deformable? If there is a reason then why the last layer only?

**Comment by forrestl111** [comment]
At first, this is only a demo shows how to use the *DeformConv2D* module. Of course, you can use deform-conv-layers to replace all the other plain layers, but it will consume more memories. Another reason is that deformable convolution offen adds to the last few conv layers of the net to better learn the offsets, and in the article,  they push the deform-conv-layers to the last three conv layers as described in 
*Section 2.3* of the article.

**Comment by bkvie** [comment]
1. I can't run more than 1 conv, not enough memory, the slicing operations are so costly. Might need to do them inplace. 
2.  Does the model allow for different sized image input?
3. The offset network:

`self.offsets = nn.Conv2d(128, 18, kernel_size=3, padding=1)`

If I use a different kernel size the output 18 would change? If I have a 5x5 kernel it would be, 5 * 5 * 2 = 50 or am I mistaken?

4. This makes me wonder how to deal with the offset then:

```
offsets = self.offsets(x)
x = F.relu(self.conv4(x, offsets))
```

**Comment by forrestl111** [comment]
1. Memory consumption is related to the size of the feature map(h*w), so when feature map is small, less memory will be cost. You can reduce the batch size to save more memory too. And, yes, more optimization should be done to save memory, could you please check the code and help optimizing it? It will be a great pleasure if you can join this project. 
2. Offsets work by these ways. First, you have to make sure the output channels' number equal to double of the size of the deformable kernel, e.g., it should be set to 2\*3\*3=18 if the kernel size in deform-conv is 3\*3. Second, the output feature map's size should match the input(output) feature map's size of the deform-conv. Besides, **stride is not available yet**, so the input and output feature maps of the deform-conv have same size.
Best regards.

**Comment by YanWang2014** [comment]
Hi, the DeformConv2D changes my feature map size from 32 to 96, what happened?
